### PR TITLE
Remove otlp_to_oc, redundant with MetricDataFromOtlp (~0 overhead) + ResourceMetricsToOC

### PR DESCRIPTION
**Pull request by bogdandrutu**

### CODE REVIEW

File: receiver/otlpreceiver/metrics/otlp.go [hunk 1]
```diff
@@ -57,8 +57,14 @@ const (
...
 	receiverCtx := obsreport.ReceiverContext(ctx, r.instanceName, receiverTransport, receiverTagValue)
 
-	for _, resourceMetrics := range req.ResourceMetrics {
-		err := r.processReceivedMetrics(receiverCtx, resourceMetrics)
+	md := data.MetricDataFromOtlp(req.ResourceMetrics)
+	rms := md.ResourceMetrics()
+	for i := 0; i < rms.Len(); i++ {
+		ocmd := internaldata.ResourceMetricsToOC(rms.Get(i))
+		if len(ocmd.Metrics) == 0 {
+			continue
```
Comment (tigrannajaryan) [inline receiver/otlpreceiver/metrics/otlp.go]: @pjanotti do we need to count this somehow in internal metrics? This is the case when we receive a request but it contains no metrics.

Comment (bogdandrutu) [reply receiver/otlpreceiver/metrics/otlp.go]: @tigrannajaryan this PR did not change this, so we probably need to file an issue to fix this.

**Comment by bogdandrutu** [comment]
@tigrannajaryan PTAL

**Comment by bogdandrutu** [comment]
Travis passes https://travis-ci.org/github/open-telemetry/opentelemetry-collector/builds/666519561?utm_source=github_status&utm_medium=notification but status does not get updated.

**Pull request merged**